File: src/main/java/org/goafabric/calleeservice/logic/CalleeLogic.java
package org.goafabric.calleeservice.logic;

import jakarta.enterprise.context.ApplicationScoped;
import org.goafabric.calleeservice.controller.Callee;

@ApplicationScoped
public class CalleeLogic {
    public Callee sayMyName(String name) {
        return new Callee("0", "Your name is: " + name);
    }

    public Callee sayMyOtherName(String name) {
        return new Callee("0", "Your other name is: " + name);
    }

    public Callee save(Callee callee) {
        return new Callee("0", "Storing your message: " + callee.message());
    }
}

File: src/main/java/org/goafabric/calleeservice/logic/CalleeLogic.kt
package org.goafabric.calleeservice.logic

import org.goafabric.calleeservice.service.Callee
import java.util.concurrent.atomic.AtomicLong
import javax.enterprise.context.ApplicationScoped

@ApplicationScoped
class CalleeLogic {
    private val sleepTime = AtomicLong(0L)

    fun sayMyName(name : String) : Callee {
        sleep()
        return Callee(message = "Your name is: ${name}")
    }

    fun sayMyOtherName(name : String ) : Callee {
        sleep()
        return Callee(message = "Your name is: ${name}")
    }

    fun setSleepTime(sleepTime : Long) : Callee {
        this.sleepTime.set(sleepTime)
        return Callee(message = "set sleepTime to: ${sleepTime}")
    }
    
    private fun sleep() {
        Thread.sleep(sleepTime.get())
    }
}